pkg/operator/controller/status.go

```diff
@@ -128,58 +129,77 @@ func computeStatusConditions(conditions []configv1.ClusterOperatorStatusConditio
 		degradedCondition.Message = "DNS namespace does not exist"
 	} else {
 		degradedCondition.Status = configv1.ConditionFalse
+		degradedCondition.Reason = "NamespaceExists"
+		degradedCondition.Message = "operand namespace exists"
 	}
-	conditions = setStatusCondition(conditions, degradedCondition)
 
+	return setOperatorStatusCondition(conditions, degradedCondition)
+}
+
+// computeOperatorProgressingCondition computes the operator's current Progressing status state.
+func computeOperatorProgressingCondition(conditions []configv1.ClusterOperatorStatusCondition, dnses []operatorv1.DNS) []configv1.ClusterOperatorStatusCondition {
 	progressingCondition := &configv1.ClusterOperatorStatusCondition{
 		Type:   configv1.OperatorProgressing,
 		Status: configv1.ConditionUnknown,
 	}
 	numDNSes := len(dnses)
-	numDaemonSets := len(daemonsets)
-	if numDNSes == numDaemonSets {
+	var dnsesAvailable int
+	for _, dns := range dnses {
+		for _, c := range dns.Status.Conditions {
+			if c.Type == operatorv1.DNSAvailable &&
+				c.Status == operatorv1.ConditionTrue {
+				dnsesAvailable++
+				break
+			}
+		}
+	}
+	if numDNSes == dnsesAvailable {
 		progressingCondition.Status = configv1.ConditionFalse
+		progressingCondition.Reason = "AllDaemonSetPodsRunning"
```

Comment:
Since I provided data on how other operators set the `Degraded` condition, I'll provide similar data on the `Progressing` condition, for your consideration:
```console
% oc get clusteroperators -o json | jq -r '.items|.[].status.conditions|.[]|select(.type=="Progressing")|[.reason][0]' | sort |
uniq -c | sort -n
      1 Ready
      1 ReconcilingComplete
     10 AsExpected
     13 null
```